src/main/scala/package.scala

```diff
@@ -1,21 +1,24 @@
 package com.originate.scalypher
 
+import com.originate.scalypher.action.ActionNodeOrRelationship
 import com.originate.scalypher.action.ActionReference
 import com.originate.scalypher.where.ObjectReference
-import com.originate.scalypher.where.Reference
+import com.originate.scalypher.where.NodeOrRelationshipReference
 
 package object types {
-  trait Referenceable {
-    def as(name: String): ActionReference =
-      ActionReference(ObjectReference(this), Some(name))
+  trait Referenceable
```

Comment:
i like renaming `Referenceable` => `Identifiable`. what are your thoughts on making this a sealed trait, having a second trait `Unreferenceable` here that `Path` extends? `sealed` is a nice mental guarantee that when we say `Identifiable` we mean something of this known set, right here, and things subtyping them.